python/flatbuffers/builder.py

```diff
@@ -441,6 +442,38 @@ def CreateByteVector(self, x):
 
         return self.EndVector(len(x))
 
+    def CreateNumpyVector(self, x):
+        """CreateNumpyVector writes a numpy array into the buffer."""
+
+        if np is None:
+            # Numpy is required for this feature
+            raise NumpyRequiredForThisFeature("Numpy was not found.")
+
+        if not isinstance(x, np.ndarray):
+            raise TypeError("non-numpy-ndarray passed to CreateNumpyVector")
+
+        if x.ndim > 1:
+            raise TypeError("multidimensional-ndarray passed to CreateNumpyVector")
+
```

Comment:
Please add an additional check that asserts that the `dtype` is a number, as well as a test for it. It's possible that users will try to pass in complicated objects (strings, etc) and we want to be clear that, right now, we won't support that.

Reply:
Done. I allow bool, int, unsigned and float elements.